packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx

```diff
@@ -0,0 +1,375 @@
+      multiline && classNames.multiline,
+      borderless && classNames.borderless,
+      underlined && classNames.underlined,
+      normalize,
+      {
+        position: 'relative',
+        selectors: {
+          [HighContrastSelector]: {
+            borderWidth: 2
+          }
+        }
+      },
+      focused && {
+        borderColor: semanticColors.inputFocusBorderAlt
+      },
+      underlined &&
+        !focused && {
+          border: `0px solid ${semanticColors.inputBorder}`
+        },
+      underlined &&
+        !disabled &&
+        !focused && {
+          selectors: {
+            ':hover': {
+              borderColor: semanticColors.inputBorderHovered
+            }
+          }
+        },
+      className
+    ],
+    wrapper: [
+      classNames.wrapper,
+      underlined && {
+        display: 'flex',
+        borderBottomWidth: 1,
+        borderBottomStyle: 'solid',
+        borderBottomColor: 'inherit',
+        width: '100%'
+      },
+      hasErrorMessage && {
+        borderColor: semanticColors.errorText,
+        selectors: {
+          '&:focus, &:hover': {
+            borderColor: semanticColors.errorText
+          }
+        }
+      },
+      hasErrorMessage &&
+        underlined &&
+        !disabled && {
+          borderBottom: `1px solid ${semanticColors.errorText}`,
+          selectors: {
+            ':focus': {
+              borderBottom: `1px solid ${semanticColors.errorText}`
+            },
+            ':hover': {
+              borderBottom: `1px solid ${semanticColors.errorText}`
+            }
+          }
+        },
+      underlined &&
+        disabled && {
+          borderBottomColor: semanticColors.disabledBackground
+        },
+      underlined &&
+        !disabled && {
+          selectors: {
+            ':hover': {
+              selectors: {
+                [HighContrastSelector]: {
+                  borderColor: 'Highlight'
+                }
+              }
+            }
+          }
+        },
+      underlined &&
+        focused && {
+          selectors: {
+            [HighContrastSelector]: {
+              borderColor: 'Highlight'
+            }
+          }
+        }
+    ],
+    fieldGroup: [
+      classNames.fieldGroup,
+      normalize,
+      {
+        border: `1px solid ${semanticColors.inputBorder}`,
+        background: semanticColors.bodyBackground,
+        height: 32,
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'stretch',
+        position: 'relative',
+        selectors: {
+          ':-ms-clear': {
+            display: 'none'
+          },
+          ':hover': {
+            selectors: {
+              [HighContrastSelector]: {
+                borderColor: 'Highlight'
+              }
+            }
+          },
+          '::placeholder': {
+            color: semanticColors.inputPlaceholderText,
+            opacity: 1
+          },
+          ':-ms-input-placeholder': {
```

Comment:
>':-ms-input-placeholder [](start = 10, length = 23)

I would actually have expected ::placeholder to include all browser variants (this and others like ::-webkit-input-placeholder) automatically hmm, and if so, this might be redundant..